lib/galaxy/managers/workflows.py

```diff
@@ -413,9 +413,21 @@ def cancel_invocation(self, trans, decoded_invocation_id):
             trans.sa_session.add(workflow_invocation)
             with transaction(trans.sa_session):
                 trans.sa_session.commit()
-        else:
-            # TODO: More specific exception?
-            raise exceptions.MessageException("Cannot cancel an inactive workflow invocation.")
+
+        for step in workflow_invocation.steps:
+            for job in step.jobs:
+                job.mark_deleted()
+                trans.sa_session.add(job)
+            if step.implicit_collection_jobs:
+                for icjja in step.implicit_collection_jobs.jobs:
+                    icjja.job.mark_deleted()
```

Comment:
I ... would maybe open a different PR and remove the option that has been deprecated for some years ?

Reply:
Sure, but that PR needs to be merged before this one if you don't make this change since the default in the method is False:
```
    def mark_deleted(self, track_jobs_in_database=False):
```
and in the config file is True.

Reply:
I don't think it matters, the handling for deleted_new and deleted is the same?